Image Classification/awesome_classifier.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+"""Image-Classification.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1StOyfUgZVCPmQWyAJCvGsMwtpe69Uky3
+"""
+
+#Loading the necessary libraries
+import os
+import numpy as np
+import pandas as pd
+import torch 
+import glob
+import torch.nn as nn
+from torchvision.transforms import transforms
+from torch.utils.data import DataLoader
+from torch.optim import Adam
+from torch.autograd import Variable
+import torchvision
+import pathlib
+
+device = torch.device("cuda" if torch.cuda.is_available() 
+                                  else "cpu")
+
+"""Data Preprocessing"""
+
+transformer = transforms.Compose([
+                                  transforms.Resize((150,150)),
+                                  transforms.RandomHorizontalFlip(),
+                                  transforms.ToTensor(), #0-255 will be changed to 0-1 and datatype will be changed from numpy to tensor.
+                                  transforms.Normalize([0.5,0.5,0.5],
+                                                       [0.5,0.5,0.5])
+                                ])
+
+from google.colab import drive
+
+drive.mount('/content/gdrive')
+
+#dataload to feed in the dataset in batch sizes.
+#Path for training and testing dataset.
+
+train_path = '/content/gdrive/My Drive/Image Classification Dataset/archive/seg_train/seg_train'
+test_path = '/content/gdrive/My Drive/Image Classification Dataset/archive/seg_test/seg_test'
+
+train_loader = DataLoader(torchvision.datasets.ImageFolder(train_path, transform=transformer),batch_size=256, shuffle=True)
+
+test_loader = DataLoader(torchvision.datasets.ImageFolder(test_path, transform=transformer),batch_size=256, shuffle=True)
+
+#categories
+
+root=pathlib.Path(train_path)
+classes = sorted([j.name.split('/')[-1] for j in root.iterdir()])
+print(classes)
+
+"""**CNN Class**"""
+
+class ConvNet(nn.Module):
+  def __init__(self,num_classes=6):
```

Comment:
The goal of this script is to use a custom trained pytorch model and give inference. Eg [This is for Keras models]:https://github.com/smaranjitghose/img_ai_app_boilerplate/blob/master/img_classifier.py. We do not need to train the models here